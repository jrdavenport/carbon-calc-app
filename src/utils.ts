export interface transportData {
  schoolName: string,
  className: string,
  zone: {
    animal: string;
    colour: string;
  };
  transport: string;
  date: string;
  journey: string;
}

export interface buttonValue {
  value: string;
  img?: string;
  colour?: string;
}

export interface school {
  name: string,
  classes: string[],
  zones: {
    animals: {
      value: string,
      img: string,
    }[],
    colours:
    {
      value: string,
      colour: string,
    }[],
  },
  transports: {
    value: string,
    img: string,
  }[],
}

export interface selectedClass {
  schoolName: string,
  className: string,
  zones: {
    animals: {
      value: string,
      img: string,
    }[],
    colours:
    {
      value: string,
      colour: string,
    }[],
  },
  transports: {
    value: string,
    img: string,
  }[],
}

export const transportStorageKey = "transportData";
export const schoolStorageKey = "schoolData";

const toLowerSnakeCase = (string: string) =>
  string.replace(" ", "_").toLowerCase();

export const adaptRecords = (records: Array<transportData>) => {
  console.log("recs", records);
  return records.map(({ journey, date, zone, transport }, index) => ({
    pupil_id: index,
    class_id: 1,
    school_id: 1,
    date,
    journey: journey.toLowerCase(),
    shares: 0,
    method: toLowerSnakeCase(transport),
    zone_animal: toLowerSnakeCase(zone.animal),
    zone_colour: toLowerSnakeCase(zone.colour),
  }));
};
