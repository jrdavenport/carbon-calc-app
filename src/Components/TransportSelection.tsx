import { Grid } from '@material-ui/core';
import React from 'react';
import SelectionButton from '../Components/SelectionButton';
import { buttonValue } from '../utils';

interface Props {
  // selectedAnimal: string;
  // selectedColour: string;
  transports: buttonValue[];
  handleNext: (selectedValue: buttonValue) => void;
}

function TransportSelection({
  // selectedAnimal,
  // selectedColour,
  transports,
  handleNext,
}: Props) {
  return <Grid container spacing={5} justify="center" >
    {transports.map((transport: buttonValue, index: number) => {
      return <Grid item xs={12} sm={3} key={index} >
        <SelectionButton text={{ transport: transport.value }} transport={transport.img} onClick={() => handleNext(transport)} />
      </Grid>
    })}
  </Grid >;
}

export default TransportSelection
