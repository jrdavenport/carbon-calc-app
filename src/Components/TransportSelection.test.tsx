import {render} from '@testing-library/react'

import React from "react";
import {buttonValue} from "../utils";
import TransportSelection from "./TransportSelection";


const setup = (mockHandleNext: (selectedValue: buttonValue) => void) => {
    const {container} = render(<TransportSelection
        handleNext={mockHandleNext}/>);
    return {
        container,
    };
};

it('matches snapshot', () => {
    const {container} = setup(() => {
    });

    expect(container).toMatchSnapshot();
})

it('calls handle next when button clicked', () => {

    let selectedButton: buttonValue | undefined = undefined

    const handleNext: (selectedValue: buttonValue) => void = (selectedValue: buttonValue) => {
        selectedButton = selectedValue
    }

    setup(handleNext)

    // Click button
    fireEvent.click(screen.getByText('Train'))

    expect(selectedButton).toEqual({"img": "ðŸš†", "value": "Train"});

})
