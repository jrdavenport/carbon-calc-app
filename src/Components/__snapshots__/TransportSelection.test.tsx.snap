// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Walk
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiTypography-root makeStyles-bigEmoji-2 MuiTypography-body1"
            >
              ðŸ¥¾
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Bike
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiTypography-root makeStyles-bigEmoji-2 MuiTypography-body1"
            >
              ðŸš²
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Bus
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiTypography-root makeStyles-bigEmoji-2 MuiTypography-body1"
            >
              ðŸšŒ
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Train
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiTypography-root makeStyles-bigEmoji-2 MuiTypography-body1"
            >
              ðŸš†
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Petrol Car
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiTypography-root makeStyles-smallEmoji-3 MuiTypography-body1"
            >
              ðŸš˜ðŸ‡µ
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Diesel Car
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiTypography-root makeStyles-smallEmoji-3 MuiTypography-body1"
            >
              ðŸš˜ðŸ‡©
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Electric Car
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiTypography-root makeStyles-smallEmoji-3 MuiTypography-body1"
            >
              ðŸš˜âš¡
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
